hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1414,9 +1421,9 @@ private void listStatusFindKeyInTableCache(
       }
       OmKeyInfo cacheOmKeyInfo = entry.getValue().getCacheValue();
       // cacheOmKeyInfo is null if an entry is deleted in cache
-      if (cacheOmKeyInfo != null) {
-        if (cacheKey.startsWith(startCacheKey) &&
-            cacheKey.compareTo(startCacheKey) >= 0) {
+      if (cacheKey.startsWith(startCacheKey) &&
+          cacheKey.compareTo(startCacheKey) >= 0) {
+        if (cacheOmKeyInfo != null) {
```

Comment:
Any reason to move this line (`if (cacheOmKeyInfo != null)`) below the prefix and compare check? This means we would be comparing keys with deleted keys and then throwing the deleted ones away. We could avoid the compare operations by discarding the deleted ones first.

Reply:
The question is are deleted keys more common than keys that do not match prefixes? I would guess in a normal load pattern having keys in the cache that are not in the prefix range is more common.

Reply:
Also, it should be cheaper to do prefix checks than look up the cache..

Reply:
At this point we already have `cacheOmKeyInfo`, comparing it to `null` is cheaper than string prefix check.

So

```java
      if (cacheOmKeyInfo != null &&
          cacheKey.startsWith(startCacheKey) &&
          cacheKey.compareTo(startCacheKey) >= 0) {
```

is cheaper than

```java
      if (cacheKey.startsWith(startCacheKey) &&
          cacheKey.compareTo(startCacheKey) >= 0 &&
          cacheOmKeyInfo != null) {
```

Reordering would make sense if we kept `deletedKeySet` and wanted to avoid growing that for keys that do not match the prefix.